src/alpaka/AlpakaCore/threadfence.h

```diff
@@ -1,48 +1,47 @@
-#ifndef HeterogeneousCoreCUDAUtilities_threadfence_h
-#define HeterogeneousCoreCUDAUtilities_threadfence_h
+#ifndef AlpakaCore_threadfence_h
+#define AlpakaCore_threadfence_h
 
 #include <type_traits>
 
-#include "AlpakaCore/alpakaCommon.h"
-#include "AlpakaCore/alpakaKernelCommon.h"
+#include "AlpakaCore/alpakaConfig.h"
 
 namespace cms::alpakatools {
 
   // device-wide memory fence
-  template <typename T_Acc>
-  ALPAKA_FN_ACC ALPAKA_FN_INLINE void threadfence(T_Acc const& acc) {
-    static_assert(std::is_same_v<T_Acc, void>,
-                  "cms::alpakatools::threadfence<T_Acc>(acc) has not been implemented for this Accelerator type.");
+  template <typename TAcc>
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE void threadfence(TAcc const& acc) {
+    static_assert(std::is_same_v<TAcc, void>,
+                  "cms::alpakatools::threadfence<TAcc>(acc) has not been implemented for this Accelerator type.");
```

Comment:
That's never supposed to happen, the point is to trigger the `static_assert` for any `TAcc` for which there is not better overload of the `threadfence` function.
Since `static_assert(false, ...)` is not allowed, and something like `std::dependent_false` never made it into the standard (?), this is  a simple way to make it always `false` while still being dependent on the `TAcc` type.

Reply:
We could replace it with `alpaka::meta::DependentFalseType<TAcc>::value` to better clarify the intent.

Reply:
On the other hand, we should actually replace the whole `cms::alpakatools::threadfence<TAcc>` with `alpaka::mem_fence` (available in `0.8.0` and later), but as it's not particularly urgent it hasn't happened yet.

Reply:
Ok, thanks for the explanation.

Reply:
See #300